packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2598,10 +2598,13 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
         }
         return true;
       }());
-      // If we're the root of the render tree (probably a RenderView),
-      // then we have to paint ourselves, since nobody else can paint
-      // us. We don't add ourselves to _nodesNeedingPaint in this
-      // case, because the root is always told to paint regardless.
+      // If we are the root of the render tree and not a repaint boundary
+      // then we have to paint ourselves, since nobody else can paint us.
+      // We don't add ourselves to _nodesNeedingPaint in this case,
+      // because the root is always told to paint regardless.
+      //
+      // Typical trees rooted at a RenderView do not go through this
```

Comment:
Nit: Should "Typical" be "Typically"?

If using "Typical": The trees are typical.

If using "Typically": The action of not going through the code path is typical.

Reply:
I have no idea which is better. Yes, normal trees we see everyday (thus typical trees) should not go to this path. And yes, the action is also seen everyday (typically)

If you like I can change it

Reply:
How about we just remove "typical" and start the sentence here with "Trees ..."?

Reply:
Done